internal/olm/operator/internal/registry_pod_test.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+
+package olm
+
+import (
+	"context"
+	"flag"
+	"path/filepath"
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+)
+
+// todo(rashmigottipati): add unit tests and use fake client
+// newFakeClient() returns a clientset but is not a replacement for a real clientset
+func newFakeClient() kubernetes.Interface {
+	return fake.NewSimpleClientset()
+}
+
+// GetKubeClient creates a kubernetes clientset using kubeconfig
+func GetKubeClient(t *testing.T) (client kubernetes.Interface) {
+	var kubeconfig *string
+	if home := homedir.HomeDir(); home != "" { // Check if machine has home directory
+		// Read kubeconfig flag. If not provided, use config file $HOME/.kube/config
+		//nolint:lll
+		kubeconfig = flag.String("kubeconfig", filepath.Join(home, ".kube", "config"), "(optional) absolute path to the kubeconfig file")
+	} else {
+		kubeconfig = flag.String("kubeconfig", "", "absolute path to the kubeconfig file")
+	}
+	flag.Parse()
+
+	// Build configuration from the config file
+	config, err := clientcmd.BuildConfigFromFlags("", *kubeconfig)
+	if err != nil {
+		t.Error(err)
+	}
+
+	// Create kubernetes clientset, which can be used to create, delete, patch, list kubernetes resources
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		t.Error(err)
+
+	}
+
+	return clientset
+}
+
+// TestCreatePod uses a fake clientset to create a pod
+func TestCreatePod(t *testing.T) {
+	rp := &RegistryPod{
+		Kubeclient: newFakeClient(),
+	}
+
+	//nolint:lll
+	_, err := rp.Kubeclient.CoreV1().Pods("default").Create(context.Background(), &corev1.Pod{ObjectMeta: metav1.ObjectMeta{Name: "testPod"}}, metav1.CreateOptions{})
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+// TestCreateBundleRegistry calls Create method which creates a bundle registry pod
+func TestCreateBundleRegistry(t *testing.T) {
+	rp := &RegistryPod{
+		Kubeclient:  GetKubeClient(t),
```

Comment:
This currently looks more like an e2e than a unit test since it's using a live cluster. Perhaps this would be a good candidate for [envtest](https://godoc.org/sigs.k8s.io/controller-runtime/pkg/envtest) testing.

Reply:
good point, removed the test case.